src/httpc.h

```diff
@@ -133,11 +133,11 @@ struct httpc_request {
 	 */
 	bool set_keep_alive_header;
 	/**
-	 * True when Transfer-Encoding: chunked must be
-	 * set before execution automatically if chunked
-	 * io enabled.
+	 * It's the number of bytes of data in the body of the
+	 * request. The value is equal to "Content-Length"
+	 * header or 0 otherwise.
 	 */
-	bool set_chunked_header;
+	long content_length;
```

Comment:
To thinkL `long` is 8 bytes on amd64 and aarch64, so everything should be OK there on the platforms we're support (at least for Linux environments). However, it looks a bit inconsistent to use `long` with `CURLOPT_INFILESIZE_LARGE`, which AFAIU is introduced after `CURLOPT_INFILESIZE` just to accept a type that is possibly wider than `long`.

Reply:
https://github.com/georgi/curl/blob/master/lib/README.curl_off_t

curl_off_t is a data type provided by the external libcurl include headers. It is the type meant to be used for the curl_easy_setopt() options that end with LARGE. The type is 64bit large on most modern platforms.

Reply:
https://github.com/curl/curl/blob/3f784980cbbea61881a30e61c86f7c32caac361a/lib/setopt.c#L1371-L1390

Reply:
Thank you, updated `long` + `CURLOPT_INFILESIZE`.

Reply:
OK, thanks! This change obsoletes current commit/PR description a bit (it mentions `CURLOPT_INFILESIZE_LARGE`).

Reply:
I'm sorry, updated.